samples/android/Continote/app/src/main/java/com/firebasecontinue/sample/continote/EditNoteActivity.java

```diff
@@ -87,7 +136,31 @@ protected void handleUserSignedIn(FirebaseUser user) {
             throw new AssertionError("user must be non-null");
         }
 
-        // TODO
+        // Try to get the current value of the Note to edit from the Firebase Realtime
+        // Database. We only get this value once here to keep things simple.
+        // The Note could change elsewhere while on this screen, thus saving could
+        // overwrite any of those changes.
+        mDatabaseRef = FirebaseDatabase.getInstance().getReference(
+                "notes/" + user.getUid() + "/" + mDatabaseKey);
+        mDatabaseRef.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(final DataSnapshot snapshot) {
+                runOnUiThread(new Runnable() {
+                    public void run() {
+                        handleNoteToEditFound(snapshot.getValue(Note.class));
```

Comment:
call getValue() _before_ you can runOnUiThread and store the result in a local, final variable. I expect that getValue likely doesn't do anything complicated, but it's always better to post only to the other thread what you actually need there.